scripts/cargo-rmc

```diff
@@ -94,6 +63,134 @@ def main():
+        config_group.add_argument("--config-toml", help="Where to read configuration from; defaults to crate's Cargo.toml")
+        config_group.add_argument("--no-config-toml", action="store_true", help="Do not use any configuration toml")
+
+        exclude_flags = [
+            # This should be able to be supported; https://github.com/model-checking/rmc/issues/359
+            "--gen-symbols",
+            # This should be able to be supported; https://github.com/model-checking/rmc/issues/360
+            "--allow-cbmc-verification-failure",
+        ]
+        rmc_flags.add_flags(parser, {"default-target": "target"}, exclude_flags=exclude_flags)
+        
+        return parser
+
+    # Check for conflicting flags
+    def validate(args):
+        rmc.ensure(not (args.crate and args.crate_flag), "Please provide a single crate to verify.")   
+        rmc.ensure(not (args.no_config_toml and args.config_toml), "Incompatible flags: --config-toml, --no-config-toml")
+
+    # Fix up args before returning
+    def post_process(args):
+        # Combine positional and flag argument for input
+        cl_args.crate = cl_args.crate or cl_args.crate_flag or "."
+
+        # --quiet overrides --verbose
+        if args.quiet:
+            args.verbose = False
+
+        # Add some CBMC flags by default unless `--no-default-checks` is being used
+        if args.default_checks:
+            rmc.add_selected_default_cbmc_flags(args)
+
+    # Combine args set by config toml and provided on command line
+    def get_combined_args(config_toml, crate, cl_args):
+        # Check that config toml exists, or set to default (<crate>/Config.toml)
+        if config_toml is not None:
+            rmc.ensure(pathlib.Path(config_toml).is_file(), f"Invalid config file: {config_toml}")
+        else:
+            crate_cargo_toml = pathlib.Path(crate).joinpath("Cargo.toml")
+            if crate_cargo_toml.is_file():
+                config_toml = crate_cargo_toml
+            else:
+                return cl_args
+        
+        # Extract flags from config toml
+        toml_args = load_toml(config_toml)
+
+        # Parse args from both toml and command line
+        # but let command line args override toml args
+        combined_args = argparse.Namespace()
+        parser.parse_args(toml_args, namespace=combined_args)
+        parser.parse_args(namespace=combined_args)
+
+        return combined_args
+
+    # Remove "rmc" from arg if invoked `cargo rmc ...`
+    if len(sys.argv) >= 2 and sys.argv[1] == "rmc":
+        del sys.argv[1]
+
+    # Parse command line args to find crate and check for config flags
+    parser = create_parser()
+    cl_args = parser.parse_args()
+    validate(cl_args)
+
+    # Load flags from config toml
+    if cl_args.no_config_toml:
+        final_args = cl_args
+    else:
+        crate = cl_args.crate or cl_args.crate_flag or "."
```

Comment:
don't you have this same bit of code somewhere else?

Reply:
Extracted to a helper function.